ignite/pkg/cosmosclient/cosmosclient.go

```diff
@@ -187,18 +194,33 @@ func WithGenerateOnly(generateOnly bool) Option {
 	}
 }
 
+// WithRPCClient sets a tendermint RPC client.
+func WithRPCClient(rpc rpcclient.Client) Option {
+	return func(c *Client) {
+		c.RPC = rpc
+	}
+}
+
+// WithWaitBlockDuration sets the wait duration for WaitForBlock methods.
+func WithWaitBlockDuration(waitBlockDuration time.Duration) Option {
```

Comment:
Instead of defining this option here should we create another option type to be passed to Wait funcs and change their function signature to accepts those options?

e.g.

```go
WaitForBlockHeight(ctx context.Context, h int64, options ...WaitOption)
```

Reply:
Good idea. Instead of a func option, which is not ideal here because there's a single option and also nothing to mutate, I chose to add a simple `timeout` parameter. 

`WaitForNextBlock`, the most handy method, doesn't take that additional parameter and keeps a constant of 10 seconds.

369dae19019c04bc77d4dfe72d78a00ab53520c8

Reply:
Forgot to remove the field... 6da9c1cd